File: examples/scala/exam1/test_valid1.scala

object test_valid1 {

  import exam1._

  val graph = begin()
    .digraph("small_graph")
      .node("A").shape("rectangle")
      .node("B").node("C").shape("doublecircle")
      .edge("A").to("B").style("dotted")
      .edge("A").and("B").to("C")
    .end()

  def main( args : Array[ String ] ) = {
    println( graph )
  }

}

File: examples/c++/exam1/test_valid1.cpp

#include "exam1.hpp"
#include <iostream>
using namespace std;
using namespace exam1;

int main() {
  auto graph = begin()
    ->digraph("a small graph")
      ->node("A")->shape("rectangle")
      ->node("B")
      ->node("C")->shape("doublecircle")
      ->edge("A")->to("B")->style("dotted")
      ->edge("A")->and_("B")->to("C")
    ->end();
  cout << *graph << endl;
  return 0;
}
